R/check_lintr.R

```diff
@@ -24,7 +24,7 @@ check_lintr <- function(x = ".", quiet = FALSE) {
   if (x$package) {
     linter <- lint_package(path = x$get_path)
   } else {
-    linter <- lint_dir(x$get_path, pattern = "\\.R(md|nw)?")
+    linter <- lint_dir(x$get_path, pattern = "\\.([rR]$|[Rr]md$|[Rr]nw$)")
```

Comment:
This should be sufficient. The R file must have an uppercase R, otherwise the fail `check_filename()`.

```suggestion
    linter <- lint_dir(x$get_path, pattern = "\\.R(md|nw)?$")
```